### ISSUE TITLE
Failed to install module salsa20-x86_64 while installing kernel 4.14.57

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#4181]
### Qubes OS version:

`R3.2`

### Affected component(s): 

kernel (module `salsa20-x86_64`)

---

### Steps to reproduce the behavior:

Use `qubes-dom0-update` to install:
- `kernel-1000:4.14.57-1.pvops.qubes.x86_64`
- `kernel-qubes-vm-1000:4.14.57-1.pvops.qubes.x86_64`

### Expected behavior:

No module installation failures.

### Actual behavior:

```Failed to install module salsa20-x86_64```

More verbose log:
```
Dependencies resolved.
=============================================================================================
 Package             Arch       Version                         Repository              Size
=============================================================================================
Installing:
 kernel              x86_64     1000:4.14.57-1.pvops.qubes      qubes-dom0-cached       46 M
 kernel-qubes-vm     x86_64     1000:4.14.57-1.pvops.qubes      qubes-dom0-cached       62 M
Removing:
 kernel              x86_64     1000:4.9.35-20.pvops.qubes      @qubes-dom0-cached     179 M
 kernel-qubes-vm     x86_64     1000:4.9.35-20.pvops.qubes      @qubes-dom0-cached     206 M

Transaction Summary
=============================================================================================
Install  2 Packages
Remove   2 Packages

Total size: 108 M
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Installing  : kernel-qubes-vm-1000:4.14.57-1.pvops.qubes.x86_64                        1/4 
mke2fs 1.42.13 (17-May-2015)
  Installing  : kernel-1000:4.14.57-1.pvops.qubes.x86_64                                 2/4 
  Erasing     : kernel-qubes-vm-1000:4.9.35-20.pvops.qubes.x86_64                        3/4 
Generating grub configuration file ...
Found theme: /boot/grub2/themes/system/theme.txt
Found linux image: /boot/vmlinuz-4.14.57-1.pvops.qubes.x86_64
Found linux image: /boot/vmlinuz-4.9.56-21.pvops.qubes.x86_64
Found initrd image: /boot/initramfs-4.9.56-21.pvops.qubes.x86_64.img
Found linux image: /boot/vmlinuz-4.9.45-21.pvops.qubes.x86_64
Found initrd image: /boot/initramfs-4.9.45-21.pvops.qubes.x86_64.img
Found linux image: /boot/vmlinuz-4.9.35-20.pvops.qubes.x86_64
done
  Erasing     : kernel-1000:4.9.35-20.pvops.qubes.x86_64                                 4/4 
Failed to install module salsa20-x86_64
Failed to install module salsa20-x86_64
Generating grub configuration file ...
Found theme: /boot/grub2/themes/system/theme.txt
Found linux image: /boot/vmlinuz-4.14.57-1.pvops.qubes.x86_64
Found initrd image: /boot/initramfs-4.14.57-1.pvops.qubes.x86_64.img
Found linux image: /boot/vmlinuz-4.9.56-21.pvops.qubes.x86_64
Found initrd image: /boot/initramfs-4.9.56-21.pvops.qubes.x86_64.img
Found linux image: /boot/vmlinuz-4.9.45-21.pvops.qubes.x86_64
Found initrd image: /boot/initramfs-4.9.45-21.pvops.qubes.x86_64.img
done
Failed to install module salsa20-x86_64
  Verifying   : kernel-1000:4.14.57-1.pvops.qubes.x86_64                                 1/4 
  Verifying   : kernel-qubes-vm-1000:4.14.57-1.pvops.qubes.x86_64                        2/4 
  Verifying   : kernel-1000:4.9.35-20.pvops.qubes.x86_64                                 3/4 
  Verifying   : kernel-qubes-vm-1000:4.9.35-20.pvops.qubes.x86_64                        4/4 

Removed:
  kernel.x86_64 1000:4.9.35-20.pvops.qubes kernel-qubes-vm.x86_64 1000:4.9.35-20.pvops.qubes

Installed:
  kernel.x86_64 1000:4.14.57-1.pvops.qubes kernel-qubes-vm.x86_64 1000:4.14.57-1.pvops.qubes

Complete!
Redirecting to '/usr/bin/dnf -q check-update' (see 'man yum2dnf')
```

### General notes:

---

**Comment by marmarek** [comment]
The place where message appears suggests the problem was with the old package, not the new one.

**Comment by andrewdavidwong** [comment]
> The place where message appears suggests the problem was with the old package, not the new one.

Oh, strange. Presumably, it's not trying to install a kernel module for the kernel it's erasing. Is the message just misleading?

**Comment by BorisKourt** [comment]
I have the same message. Can this be ignored for the moment?

**Comment by esote** [comment]
@andrewdavidwong @marmarek  Looks like the change is in stable now, can this issue be closed, or is there more left to do?

**Comment by andrewdavidwong** [comment]
> Looks like the change is in stable now, can this issue be closed, or is there more left to do?

Normally, this would be closed from a commit. So, either there's still more to do, or someone forgot to include the appropriate close command in a commit message.

**Comment by esote** [comment]
marmarek did put "Fixes QubesOS/qubes-issues#4181" in the commit message, but for some reason it didn't close this issue like it normally would

**Comment by andrewdavidwong** [comment]
> marmarek did put "Fixes #4181" in the commit message, but for some reason it didn't close this issue like it normally would

It won't close this issue until the commit is merged into the repo's default branch (`master`):

https://help.github.com/articles/closing-issues-using-keywords/

You can see that if you hover over the "i" icon next to the commit line above.

**Comment by marmarek** [comment]
Unfortunately related fix in master branch lack "Fixes" tag, so it won't close this issue automatically.
It's here: QubesOS/qubes-core-admin-linux@68dd0135855aec06e9b9981dbed80f9b08aa07cc
